src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/rewriting/rules/RuleChangeOrderOfSubPlansOfMultiwayUnion.java

```diff
@@ -0,0 +1,55 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting.rules;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.NaryLogicalOp;
+import se.liu.ida.hefquin.engine.queryplan.utils.PhysicalPlanFactory;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting.RuleApplication;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RuleChangeOrderOfSubPlansOfMultiwayUnion extends AbstractRewritingRuleImpl{
+
+    public RuleChangeOrderOfSubPlansOfMultiwayUnion( final double priority ) {
+        super(priority);
+    }
+
+    @Override
+    protected boolean canBeAppliedTo( final PhysicalPlan plan ) {
+        final PhysicalOperator op = plan.getRootOperator();
+        if ( IdentifyLogicalOp.matchMultiwayUnion(op) ) {
+            for ( int i = 0; i < plan.numberOfSubPlans(); i ++ ) {
+                final PhysicalOperator subOp = plan.getSubPlan(i).getRootOperator();
+                if ( IdentifyLogicalOp.matchUnion(subOp) ) {
+                    return true;
+                }
+            }
```

Comment:
Similar question here. Why look at the subplans if the purpose of this rule is to simply change their order?

Reply:
RuleMergeChildUnionIntoMultiwayUnion